lib/interceptor/retry.js

```diff
@@ -0,0 +1,18 @@
+'use strict'
+const RetryHandler = require('../handler/retry-handler')
+
+module.exports = globalOpts => {
+  return dispatch => {
+    return function retryInterceptor (opts, handler) {
+      opts.retryOptions = { ...globalOpts, ...opts.retryOptions }
```

Comment:
I don't think we need to mutate opts here. Just make a rw copy to pass to the handler.